keps/sig-storage/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,590 @@
+- Two versions passed since introducing the functionality which deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include [conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+### Upgrade / Downgrade Strategy
+
+There is a new field getting added to the StatefulSet. The upgrade will not 
+change the previously expected behaviour of existing Statefulset. 
+
+If the statefulset had been set with the RemoveOnStatefulSetDeletion 
+and RemoveOnScaleDown and the version of the kube-controller downgraded,
+even though the `PersistentVolumeClaimReclaimPolicy` field will go away, the references
+would still be acted upon by the garbage collector and cleaned up 
+based on the settings before downgrade. 
+
+<!--
+If applicable, how will the component be upgraded and downgraded? Make sure
+this is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
+-->
+
+### Version Skew Strategy
+There is only kubecontroller manager changes involved, hence not applicable for
+version skew involving other components.
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: statefulset-autodelete-pvcs
+    - Components depending on the feature gate: kube-controller-manager
+  
+* **Does enabling the feature change any default behavior?**
+  The default behaviour is only changed when user explicitly specifies the `PersistentVolumeClaimReclaimPolicy`. 
+  Hence no change in any user visible behaviour change by default.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?**
+  Yes, but with side effects for users who already started using the feature by means of 
```

Comment:
I think it will be sufficient to say yes here, and refer to the downgrade section above on what happens on rollback.

I think that disabling the feature would also mean removing the new policy field?

I'm not sure what you mean by annotating the PVCs here.